File: frontend/src/types/deadLetterEvent.ts
import type { EventAttemptItem, EventItem, EventLogItem } from "./event";

export type DeadLetterEventItem = {
  id: string;
  event_id: string;
  reason: string;
  error_message: string | null;
  retry_count: number;
  original_routing_key: string | null;
  created_at: string;
  correlation_id: string;
  trace_id: string;
  event_status: string;
};

export type DeadLetterEventDetail = {
  id: string;
  event_id: string;
  reason: string;
  payload: Record<string, unknown>;
  retry_count: number;
  original_routing_key: string | null;
  error_message: string | null;
  created_at: string;
  event: EventItem;
  attempts: EventAttemptItem[];
  logs: EventLogItem[];
};

export type DeadLetterReprocessResponse = {
  dead_letter_event_id: string;
  event_id: string;
  status: string;
  routing_key: string;
  correlation_id: string;
  trace_id: string;
};

File: backend/app/schemas/dead_letter_event.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict

from app.schemas.event import EventAttemptRead, EventLogRead, EventRead


class DeadLetterEventListItem(BaseModel):
    id: str
    event_id: str
    reason: str
    error_message: str | None
    retry_count: int
    original_routing_key: str | None
    created_at: datetime
    correlation_id: str
    trace_id: str
    event_status: str


class DeadLetterEventDetail(BaseModel):
    id: str
    event_id: str
    reason: str
    payload: dict[str, Any]
    retry_count: int
    original_routing_key: str | None
    error_message: str | None
    created_at: datetime
    event: EventRead
    attempts: list[EventAttemptRead]
    logs: list[EventLogRead]


class DeadLetterReprocessResponse(BaseModel):
    dead_letter_event_id: str
    event_id: str
    status: str
    routing_key: str
    correlation_id: str
    trace_id: str
